src/l500/l500-options.cpp

```diff
@@ -0,0 +1,177 @@
+//// License: Apache 2.0. See LICENSE file in root directory.
+//// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include "l500-options.h"
+#include "l500-private.h"
+#include "l500-depth.h"
+
+const std::string MIN_CONTROLS_FW_VERSION("1.3.8.0");
+
+namespace librealsense
+{
+    using namespace ivcam2;
+
+    float l500_hw_options::query() const
+    {
+        return query(_resolution->query());
+    }
+
+    void l500_hw_options::set(float value)
+    {
+        _hw_monitor->send(command{ AMCSET, _type, (int)value });
+    }
+
+    option_range l500_hw_options::get_range() const
+    {
+        return _range;
+    }
+
+    l500_hw_options::l500_hw_options(hw_monitor* hw_monitor, l500_control type, option* resolution)
+        :_hw_monitor(hw_monitor),
+        _type(type),
+        _resolution(resolution)
+    {
+        auto min = _hw_monitor->send(command{ AMCGET, _type, get_min });
+        auto max = _hw_monitor->send(command{ AMCGET, _type, get_max });
+        auto step = _hw_monitor->send(command{ AMCGET, _type, get_step });
+
+        auto def = query(_resolution->query());
+
+        _range = option_range{ float(*(reinterpret_cast<int*>(min.data()))),
+            float(*(reinterpret_cast<int*>(max.data()))),
+            float(*(reinterpret_cast<int*>(step.data()))),
+            def };
+    }
+
+    void l500_hw_options::enable_recording(std::function<void(const option&)> recording_action)
+    {}
+
+    float l500_hw_options::query(int mode) const
+    {
+        auto res = _hw_monitor->send(command{ AMCGET, _type, get_current, mode });
+        auto val = *(reinterpret_cast<int*>((void*)res.data()));
```

Comment:
Is there a check that the result is valid ?

Reply:
yes on send(command)